outline/shadowsocks/config.go

```diff
@@ -35,16 +42,47 @@ type Client struct {
 	client.Client
 }
 
+func extractPrefixBytes(prefixUtf8 string) ([]byte, error) {
+	// The prefix is an 8-bit-clean byte sequence, stored in the codepoint
+	// values of a unicode string, which arrives here encoded in UTF-8.
+	prefixRunes := []rune(prefixUtf8)
+	prefixBytes := make([]byte, len(prefixRunes))
+	for i, r := range prefixRunes {
+		if (r & 0xFF) != r {
+			return nil, fmt.Errorf("character out of range: %d", r)
```

Comment:
Just curious, will the shell (used by electron) or TypeScript support passing these special characters like `nul (0x00)` or `0xFE/0xFF` (which is not a valid byte in UTF-8 encoding).

Reply:
Yes, I believe this works today in v1.9.0/1.  Strictly speaking, I believe there is no "shell" between Node and Go.  I've confirmed that Node's `child_process` package formats string arguments in UTF-8 when passing them to the child process, and this step reverses the UTF-8 encoding to recover the codepoint (i.e. "rune") values.